File: golang/go_latest_version.py
# -*- coding: utf-8 -*-
# original: https://raw.githubusercontent.com/UedaTakeyuki/slider/master/mh_z19.py
#
# © Takeyuki UEDA 2020 -

import requests
from bs4 import BeautifulSoup

res = requests.get('https://golang.org/dl/')
soup = BeautifulSoup(res.text, 'html.parser')

version = soup.find('div', class_='toggleVisible')['id']

print(version)

# Beautiful Soup reference
# https://qiita.com/Chanmoro/items/db51658b073acddea4ac

File: golang/go_latest_version.go
package main

import (
	"fmt"
	"os"

	"github.com/anaskhan96/soup"
)

func main() {
	resp, err := soup.Get("https://golang.org/dl/")
	if err != nil {
		os.Exit(1)
	}
	doc := soup.HTMLParse(resp)
	//	links := doc.Find("div", "id", "comicLinks").FindAll("a")
	//	for _, link := range links {
	//		fmt.Println(link.Text(), "| Link :", link.Attrs()["href"])
	//	}
	//	version := soup.find('div', class_='toggleVisible')['id']
	version := doc.Find("div.toggleVisible").Attrs()["id"]
	fmt.Println(version)
}
